Sming/Arch/Esp8266/Components/esp8266/esp_cplusplus.cpp

```diff
@@ -11,34 +11,9 @@ extern void (*__init_array_end)();
 void cpp_core_initialize()
 {
     void (**p)(void);
-    for (p = &__init_array_start; p != &__init_array_end; ++p)
+    for (p = &__init_array_start; p != &__init_array_end; ++p) {
```

Comment:
I note in the esp8266-arduino core constructors are called in reverse order. Not sure why...

Reply:
@igrr @earlephilhower is there a reason for calling the C++ constructors in reverse order in `esp8266-arduino` ?

Reply:
I believe that was in @igrr's original work and not something I've touched, sorry.  Every example I see (well, they all look cut-n-pasted from the same source) does the initialization from [0]..[n].

While between translation units there's no guarantees, I would have thought that w/in a single file the order would be guaranteed.  May be a bug in the `esp8266-arduino` core.  Will open an issue there and we'll look into it.

Reply:
As per https://github.com/esp8266/Arduino/issues/6870 reverse-initialiation _is_ still required to ensure correct init priority. Will add test case to HostTests and fix.